### ISSUE TITLE
test_image_palette fails with Pillow 9.1.0

### ISSUE THREAD

**Issue by dotlambda** [pikepdf/pikepdf#328]
```
============================= test session starts ==============================
platform linux -- Python 3.10.3, pytest-7.0.1, pluggy-1.0.0
rootdir: /build/source, configfile: pyproject.toml, testpaths: tests
plugins: hypothesis-6.38.0, xdist-2.5.0, forked-1.4.0
gw0 [510] / gw1 [510]m
....................................................................x... [ 14%]
..........................F.................s........................... [ 28%]
....................................................s................... [ 42%]
........................................................................ [ 56%]
........................................................................ [ 70%]
....................................s................................... [ 84%]
...................................ssss................................. [ 98%]
......                                                                   [100%]
=================================== FAILURES ===================================
_________________ test_image_palette[pal-1bit-rgb.pdf-1-rgb2] __________________
[gw0] linux -- Python 3.10.3 /nix/store/443qk9752l44kfp8pzsrp0m9jyq6jz2p-python3-3.10.3/bin/python3.10

resources = PosixPath('/build/source/tests/resources')
filename = 'pal-1bit-rgb.pdf', bpc = 1, rgb = (255, 128, 0)

    @pytest.mark.parametrize(
        'filename,bpc,rgb',
        [
            ('pal.pdf', 8, (0, 0, 255)),
            ('pal-1bit-trivial.pdf', 1, (255, 255, 255)),
            ('pal-1bit-rgb.pdf', 1, (255, 128, 0)),
        ],
    )
    def test_image_palette(resources, filename, bpc, rgb):
        pdf = Pdf.open(resources / filename)
        pim = PdfImage(next(iter(pdf.pages[0].images.values())))

        assert pim.palette[0] == 'RGB'
        assert pim.colorspace == '/DeviceRGB'
        assert not pim.is_inline
        assert pim.mode == 'P'
        assert pim.bits_per_component == bpc

        outstream = BytesIO()
        pim.extract_to(stream=outstream)

        im = pim.as_pil_image().convert('RGB')
>       assert im.getpixel((1, 1)) == rgb
E       assert (255, 255, 255) == (255, 128, 0)
E         At index 1 diff: 255 != 128
E         Use -v to get the full diff

tests/test_image_access.py:520: AssertionError
=============================== warnings summary ===============================
tests/test_pages.py:220
tests/test_pages.py:220
  /build/source/tests/test_pages.py:220: PytestUnknownMarkWarning: Unknown pytest.mark.timeout - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.timeout(1)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/test_image_access.py::test_image_palette[pal-1bit-rgb.pdf-1-rgb2]
======= 1 failed, 501 passed, 7 skipped, 1 xfailed, 2 warnings in 16.94s =======
```

**Comment by jbarlow83** [comment]
Looks like this new release of Pillow is disregarding the image's palette.

**Comment by mara004** [comment]
Is the image a GIF? If so, I think the test failure may be caused by [this change](https://github.com/python-pillow/Pillow/pull/5857) in Pillow. Since version 9.1, there's a [setting](https://github.com/python-pillow/Pillow/pull/6150) to configure the behaviour.

**Comment by jbarlow83** [comment]
Fixed in v5.1.2. 

Notes: did not fully investigate if this was a Pillow _problem_ or just a behavior quirk. Regardless the fix works with new and old Pillow.
